### ISSUE TITLE
Enhancement - Course Level Progress

### ISSUE THREAD

**Issue by joe-replin** [adaptlearning/adapt-contrib-pageLevelProgress#167]
### Subject of the issue/enhancement/features
I have received feedback from a large set of user survey data that the current PLP settings give learners the false impression they have nearly completed the course. In actuality, they are almost done with the current page. I will be evaluating the recent addition `_showAtCourseLevel`, in a live setting. But additionally, I am adding a course level progress meter in the menu header that may trigger the PLP drawer. 

If there is an appetite for this visual indicator, I am happy to add my work to a PR. I believe even screen reader users may benefit from something like this.

Lastly, this plug-in now adds progress meters to various elements through the course experience and is not solely focused on a page-level experience anymore. I would suggest changing the name to **adapt-contrib-progression**.

### Rough Visual Draft
![course-progress](https://user-images.githubusercontent.com/7740674/225808667-adc4c515-a1d2-4ec4-ad22-4f2f21575cb1.png)

**Comment by oliverfoster** [comment]
We used to have a progress bar under the navigation bar. Almost as a bottom border to the navbar.

**Comment by guywillis** [comment]
Running the risk of side tracking the issue, perhaps PLP is becoming too bloated with all of the features and instead needs to be chopped up into smaller segments:

* PLP (or something similarly named) would be in control of tracking completion
* Breadcrumb progress (would happily take other suggestions) would control the display of that entire pages completion using the PLP plugin for the info
* Title Indicator or similar could control the display of the title level indicators that appear globally  

The separation of tracking progress to visual display and functionality should open up new visual ways of displaying completion data and be less locked down. Furthermore, it would open up potential for a course-wide navigational menu that's separate, but similar, to how PLP currently behaves but also allow for completion indicators to be loaded into there using the completion data from PLP.

Am keen to hear thoughts on the ideas above

**Comment by oliverfoster** [comment]
I agree about the monolithic nature of plp.

Just a reminder: the menu and page `display completion`, i.e. the representative ticks in the plp indicators, are not the same as `absolute completion`, which is the relationship between _isComplete and _isOptional, assessment completion, tracking config etc. The `_pageLevelProgress._isEnabled` property specifically indicates which items are to be included in the `display completion` and there are a series of extra calculations and flags for assessment criteria and page completion where `display completion` and `absolute completion` don't align.

Exposing the plp calculations for menus and pages should be relatively straighforward. Separating out into breadcrumb and title indicators is not sensible at this stage, as plugin dependencies are not definable. I would rather we add breadcrub to plp to break it up later.

**Comment by joe-replin** [comment]
Closing this in favor of an alternate approach.
